examples/angular/src/app/lib/resolver/dotcms-page.resolver.ts

```diff
@@ -0,0 +1,59 @@
+import { inject } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  ResolveFn,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { Observable, from, map, tap } from 'rxjs';
+
+import { DOTCMS_CLIENT_TOKEN } from '../dotcms-client-token';
+import { DotCMSNavigationItem, DotCMSPageAsset } from '../models';
+import { PageContextService } from '../services/dotcms-context/page-context.service';
+
+/**
+ * This resolver is used to fetch the page and navigation data from dotCMS.
+ *
+ * @param {ActivatedRouteSnapshot} route
+ * @param {RouterStateSnapshot} _state
+ * @return {*}
+ */
+export const DotCMSPageResolver: ResolveFn<
+  Observable<{
+    pageAsset: DotCMSPageAsset;
+    nav: DotCMSNavigationItem;
+  }>
+> = (route: ActivatedRouteSnapshot) => {
+  // Get the Service
+  const client = inject(DOTCMS_CLIENT_TOKEN);
+  const pageContextService = inject(PageContextService);
+
+  const url = route.url.map((segment) => segment.path).join('/');
+  const queryParams = route.queryParams;
+
+  const pageProps = {
```

Comment:
and pass all of this to that service.

Reply:
I'm getting those props from the router. Does it make sense to move it from a different service? I mean, we are in a resolver, we took advantage of that

Reply:
clarified @rjvelazco 🤛🏼